### PR TITLE
fix(cadl): 3rd party exec error should be returned

### PR DESCRIPTION
**Pull request by archerzz**
## Description

PowerShell does not stop and throw error if 3rd party exec returns error exit code. Some places in our CADL build scripts do not check the exit code of 3rd party exec, so that the build process will not stop on error.

### CODE REVIEW

File: eng/scripts/Cadl-Project-Generate.ps1
Comment (chunyu3) [inline eng/scripts/Cadl-Project-Generate.ps1:59]: Better to give out an error message.

**Review by chunyu3 — APPROVED** [review]

**Comment by archerzz** [comment]
wait until the fix for `adp` is merged

File: eng/scripts/Cadl-Project-Generate.ps1
Comment (archerzz) [reply eng/scripts/Cadl-Project-Generate.ps1:66]: The `npx` command will print error message, so here I just exit.

**Pull request merged**
